packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -0,0 +1,82 @@
+import { readdir } from "fs";
+import path, { dirname } from "path";
+import vscode from "vscode";
+import { Executable } from "vscode-languageclient/node.js";
+import { StartFileName } from "./const.js";
+import logger from "./log/logger.js";
+import { isFile, loadModule, normalizeSlash } from "./utils.js";
+
+export const toOutput = (str: string) => {
+  str
+    .trim()
+    .split("\n")
+    .forEach((line) => logger.info(line));
+};
+export const toError = (str: string) => {
+  str
+    .trim()
+    .split("\n")
+    .forEach((line) =>
+      logger.error(line, [], {
+        showOutput: true,
+        showPopup: false,
+      }),
+    );
+};
+
+export async function resolveTypeSpecCli(absolutePath: string): Promise<Executable | undefined> {
+  if (!path.isAbsolute(absolutePath) || (await isFile(absolutePath))) {
+    return undefined;
+  }
+  const modelInfo = await loadModule(absolutePath, "@typespec/compiler");
+  if (modelInfo) {
+    const cmdPath = path.resolve(modelInfo.path, "cmd/tsp.js");
+    return {
+      command: "node",
+      args: [cmdPath],
+    };
+  }
+  return undefined;
+}
+
+export async function getMainTspFile(tspPath: string): Promise<string | undefined> {
```

Comment:
the returned file may not be "main.tsp", suggest to rename it to getEntrypointTspFile